DDD-4.md

```diff
@@ -0,0 +1,348 @@
+}
+```
+
+The following refers to the offsets during snapshots:
+```json
+{
+  "snapshot": true, // boolean
+  // See common attributes shown above
+}
+```
+
+The following refers to the offsets during streaming:
+```json
+{
+  // See common attributes shown above
+  // See Transaction Monitoring for common transaction-based key/value pairs
+  // See Incremental Snapshots for common incremental-snapshot key/value pairs
+}
+```
+
+### Oracle
+
+The Oracle connector maintains offsets that align with our relational connectors. 
+The offsets consist of a set of state that differs based on whether the connector is in a snapshot or streaming phase.
+
+This connector maintains a single partition of offsets, based on either the configured `pdb.name` or `database.dbname`.
+
+The following refers to the offsets during snapshot:
+```json
+{
+  "scn": "<system-change-number>", // string, may be null
+  "snapshot": true, // boolean
+  "snapshot_completed": false, // boolean
+  "snapshot_pending_tx": "<tx1>:<scn1>,<tx2>:<scn2>,...", // string, comma-separated of "transaction-id:system change number" tuples
+  "snapshot_scn": "<system-change-number>" // string, may be null
+}
+```
+
+The following refers to the offsets during streaming:
+```json
+{
+  "snapshot_scn": "<system-change-number>", // string, may be null
+  "snapshot_pending_tx": "<tx1>:<scn1>,<tx2>:<scn2>,...", // string, comma-separated of "transaction-id:system change number" tuples
+  
+  // The following are applicable only if the connector is using XStream
+  "lcr_position": "<position>", // string, can be omitted if null
+  
+  // The following are applicable only if the connector is using LogMiner
+  "scn": "<system-change-number>", // string
+  "commit_scn": "<commit-scn-serialized-string>", // string, see below for serialized format specifications
+
+  // See Transaction Monitoring for common transaction-based key/value pairs
+  // See Incremental Snapshots for common incremental-snapshot key/value pairs 
+}
+```
+
+The `commit_scn` field is required by the Oracle connector when using the LogMiner adapter (the default).
+This field has undergone a significant transformation over the past several releases and therefore its critical to understand how to serialize/deserialize this as indicated below:
+
+| Version           | Format                                                                     |
+|-------------------|----------------------------------------------------------------------------|
+| Up to 1.9.4.Final | `<commit-scn>`                                                             |
+| 1.9.5.Final       | `<commit-scn>:<rollback-segment-id>:<sql-sequence-number>:<arch-thread-#>` |
+| 1.9.6.Final       | `<commit-scn>:<arch-thread-#>:<transaction-id-list>`                       |
+| 1.9.7.Final       | `<commit-scn>:<arch-thread-#>:<transaction-id-list>`                       |  
+| 2.0.0.Final+      | `<commit-scn>:<arch-thread-#>:<transaction-id-list>`                       |
```

Comment:
I suppose we are only going to support only 2.x offsets.

Reply:
Most likely, but I added this mainly for completeness.